flaml/nlp/hpo/get_grid_search_space.py

```diff
@@ -0,0 +1,437 @@
+# lookup table for the grid configs in each pre-trained language huggingface for different tasks
+import copy
+
+def get_space_union_and_unique(search_space_common, search_space_unique, this_case_tags: list):
+    search_space_union = copy.deepcopy(search_space_common)
+    this_search_space = copy.deepcopy(search_space_common)
```

Comment:
why do we need deepcopy?

Reply:
This function returns two distinct dict: search_space_union and this_search_space, the former is the search space containing all configs for each condition; the latter is the search space containing only 1 config for one specific condition. Since the two dicts are supposed to be different, we need deepcopy

Reply:
why deepcopy instead of copy?